tests/test_project_structure.py

```diff
@@ -35,3 +70,50 @@ def assert_file_not_contains(self, filename, pattern):
         with open(filename, 'rb', 0) as file, mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as content:
             if content.find(bytes(pattern, 'utf-8')) != -1:
                 self.fail(f"File {filename} contians illegal pattern - {pattern}")
+
+    def test_providers_modules_should_have_tests(self):
```

Comment:
Nice, looks good to me. Even though the code is well commented, perhaps add an explanatory docstring e.g.?
```suggestion
    def test_providers_modules_should_have_tests(self):
        """Assert every module in /airflow/providers has a corresponding test_ file in tests/airflow/providers."""
```

Reply:
Updated! Thank you!